extensions/azure/azure-test/src/testFixtures/java/org/eclipse/dataspaceconnector/azure/testfixtures/TerraformOutputsExtension.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.azure.testfixtures;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.common.testfixtures.TestUtils;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * A JUnit extension for setting system properties from a Terraform outputs JSON file.
+ * Used by cloud integration tests.
+ */
+public class TerraformOutputsExtension implements BeforeAllCallback, AfterAllCallback {
```

Comment:
Relying on a particular deployment mechanism (i.e. Terraform) and its artifacts (the json file) seems like an anti-pattern. Why can't we use Azure ResourceManager to deploy everything?   
Or, at least [run terraform from withing code](https://github.com/microsoft/terraform-spring-boot)

Reply:
That is explained in the design document, it is preferable to have a service identity with the minimal required privileges so that even if it leaks it can't be abused to create e.g. VMs.

We could have a script that converts the Terraform output json file into a key=value .env type file.

Reply:
I don't quite understand. Am I correct in assuming that the permissions/privileges would be the same regardless whether we provision resources with Terraform or the Azure ResourceManager SDK? So we need elevated privileges for resources deployment anyway. 

That would mean that during test setup (=deploying resources) a different identity should be used than during test execution, right?

However, I do understand that rebuilding the Terraform config in Java Code (Azure RM) is probably not worth the effort.

Reply:
> That would mean that during test setup (=deploying resources) a different identity should be used than during test execution, right?

Indeed. During deployment you use the developer's own identity (user), while during test you use also the developer's identity when running on your machine, and a service principal with limited permissions in CI.

> However, I do understand that rebuilding the Terraform config in Java Code (Azure RM) is probably not worth the effort.

That would not be much of an issue actually. We deploy very few resources and it could be ported to Java easily (assuming no bad surprises), if that's the way forward we choose.

Using scripts to deploy does make the dev setup more complex and potentially less portable (to Windows). On the other hand, having the resources already deployed and their configuration downloaded locally makes running the test many times for developers ("inner loop") much faster than having the overhead of running Terraform each time. We might have a "best of both worlds scenario" with a Java test framework that "caches" the configuration locally between runs.

Reply:
@paullatzelsperger we have updated it to use `.properties` file and removed use of terraform json artifact.